### PR TITLE
fixed the bug where continue backwards did nothing.

### LINKED ISSUE #14: Continuing backwards does nothing.
This issue relates to the _Continue backwards_ functionality. Clicking on the button does nothing and the current instruction pointer stays the same.

Testing was done on the following:
**Ida Pro 7.7
Windows 10
WinDbg engine version: 10.0.25877.1004
WinDbg client version: 1.2306.12001.0**

 A GIF of the issue is attached below.
Animation

### PR DESCRIPTION
**Pull request by japjitsingh**
This commit resolves #14. 

The bug relates to calling _Cursor::ReplayBackward_ with -1 for the _stepCount_ argument. The C++ code examples in the _ttd-bindings_ repository mistakenly calls _Cursor::ReplayForward_ with -1. It should be called with -2. Although _Cursor::ReplayForward_  works normally, using the same argument for _Cursor::ReplayBackward_ does nothing.

The fix can be confirmed by looking at the code for Python bindings in the [ttd-bindings repository.](https://github.com/commial/ttd-bindings/blob/master/python-bindings/module.cpp)

It correctly adds the following line:
`m.attr("MAX_STEP") = 0xFFFFFFFFFFFFFFFE;`
which as an unsigned long equates to -2.

Any calls to continue code execution is then achieved by using _pyTTD.MAX_STEP_ as seen in [example_api.py.](https://github.com/commial/ttd-bindings/blob/master/example_api/example_api.py)

### CODE REVIEW

**Comment by japjitsingh** [comment]
Hello!

I accidentally reverted my second commit, so we see 3 new commits. (2nd commit, revert, 2nd commit)
Sorry about that. 

The latest commit resolves #12 which occurs when stepping backwards, and mildly improves code readability and design.

As per my understanding of the plugin, when stepping backwards:
- TTD Engine is used to go one instruction back.
- A Breakpoint event is raised to signal to IDA to update its UI. 
- IDA complains about an unknown breakpoint as there is no breakpoint at the arrived-at instruction.

In terms of original code implementation, the stepping backwards functionality is achieved by the following steps:
 - **BackwardSingleStepRequest::activate()** sets **m_backwardsSingleStep** to true and calls **continue_process()**. 
 - The event-handler **DebuggerManager::onResume()** is triggered:
     - If **m_backwardsSingleStep** is true, **m_isForward** is set to false, and **applyCursor(1)** is called.
     - A breakpoint event is raised which simulates hitting of a breakpoint at the arrived-at instruction.
     - The original value of **m_isForward** is restored, **m_backwardsSingleStep** is set to false (as its default value) and the function returns.
- **BackwardSingleStepRequest::activate()** returns.
    
The code centering around **m_backwardsSingleStep** seems a bit of a "hack". I have introduced some changes to the code that eliminates **m_backwardsSingleStep** while keeping all functionality intact.

The changed code flow is as follows:
- **BackwardSingleStepRequest::activate()** toggles **m_isForward**, explicitly sets **m_resumeMode** to **resume_mode_t::RESMOD_INTO** and calls **step_into()**.
  - The event-handler **DebuggerManager::onResume()** is triggered:
    - The switch statement matching **resume_mode_t::RESMOD_INTO** is taken, thereby calling **applyCursor(1).**
    - A step event is raised.
    - **m_resumeMode** is set back to its default value of **resume_mode_t::RESMOD_NONE**, and the function returns.
- **BackwardSingleStepRequest::activate()** toggles **m_isForward** again and returns.

Some other changes that were made to ensure proper functionality:
- **BackwardStateRequest::activate()** explicitly sets **m_resumeMode** to **resume_mode_t::RESMOD_NONE**.
    - This is technically not needed as **m_resumeMode** is previously set to **resume_mode_t::RESMOD_NONE** anyway.
      - Only done to improve code comprehension.

In conclusion: the commit gets rid of *m_backwardsSingleStep* and significantly simplifies the *DebuggerManager::onResume()* function. Instead of relying on a boolean value, backward-step-into simply changes the direction of the execution and uses the same code as that of the normal step-into. Breakpoint event is not used and we will not get the "unknown debugger breakpoint" errors while stepping backwards.

Thank you for the great plugin, and please keep up the astounding work! :grin:

Japjit
